src/SparkTestBase.sol

```diff
@@ -161,6 +162,35 @@ abstract contract SparkTestBase is ProtocolV3TestBase {
         }
     }
 
+    function testTokenImplementationsMatch() public {
+        // This test is to avoid a footgun where the token implementations are upgraded (possibly in an emergency) and
+        // the config engine is not redeployed to use the new implementation. As a general rule all reserves should
+        // use the same implementation for AToken, StableDebtToken and VariableDebtToken.
+        GovHelpers.executePayload(vm, payload, executor);
+
+        IPoolConfigurator poolConfigurator = IPoolConfigurator(poolAddressesProvider.getPoolConfigurator());
+        address[] memory reserves = pool.getReservesList();
+        assertGt(reserves.length, 0);
+
+        DataTypes.ReserveData memory data = pool.getReserveData(reserves[0]);
+        address aTokenAddress            = getImplementation(address(poolConfigurator), data.aTokenAddress);
+        address stableDebtTokenAddress   = getImplementation(address(poolConfigurator), data.stableDebtTokenAddress);
+        address variableDebtTokenAddress = getImplementation(address(poolConfigurator), data.variableDebtTokenAddress);
```

Comment:
```suggestion
        address aTokenImplAddress            = getImplementation(address(poolConfigurator), data.aTokenAddress);
        address stableDebtTokenImplAddress   = getImplementation(address(poolConfigurator), data.stableDebtTokenAddress);
        address variableDebtTokenImplAddress = getImplementation(address(poolConfigurator), data.variableDebtTokenAddress);
```

Reply:
OR
```suggestion
        address aTokenImplementationAddress            = getImplementation(address(poolConfigurator), data.aTokenAddress);
        address stableDebtTokenImplementationAddress   = getImplementation(address(poolConfigurator), data.stableDebtTokenAddress);
        address variableDebtTokenImplementationAddress = getImplementation(address(poolConfigurator), data.variableDebtTokenAddress);
```

Reply:
It'd do this, because now the variable names introduce a lot of confusion on what is an implementation address and what is an actual address

Reply:
yes good point - replaced address with impl. fixed.